hypershift-operator/main.go

```diff
@@ -246,6 +248,27 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 		}
 	}
 
+	// The mgr and therefore the cache is not started yet, thus we have to construct a client that
+	// directly reads from the api.
+	apiReadingClient, err := crclient.NewDelegatingClient(crclient.NewDelegatingClientInput{
+		CacheReader: mgr.GetAPIReader(),
+		Client:      mgr.GetClient(),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to construct api reading client: %w", err)
+	}
+
+	// Convert RegistryOverrides flag to a map of string-to-slice of strings
+	registryOverrides := util.ConvertRegistryOverrideFlagDataType(opts.RegistryOverrides)
+
+	// Populate RegistryOverrides with ICSP from management cluster if no value is passed in the RegistryOverrides flag
+	if len(registryOverrides) == 0 {
```

Comment:
what I was thinking here is that you populate a set of registry overrides from whatever is in the ImageContent policy of the mgmt cluster and then apply what was specified in the command flag on top of that. That way the flag can overwrite something on the mgmt cluster but both are picked up.

Reply:
Ok no issues changing this. I thought we originally discussed if they didn't pass anything in, use the mgmt cluster, but if they did pass something in, use their overrides.